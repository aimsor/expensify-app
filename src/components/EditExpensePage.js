import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

// Refactor EditExpensePage to be a class based component
// Setup mapDispatchToProps: editExpense and removeExpense
// dev mode
// done now

// should render EditExpensePage
// snapshot

// should handle editExpense
// spies

// should handle removeExpense
// spies

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button
          onClick={this.onRemove}
        >Remove</button>
      </div> 
    );
  };
};

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expense) => {
//           console.log('updated', expense);
//           props.dispatch(editExpense(props.expense.id, expense));
//           props.history.push('/');
//         }}
//       />
//       <button
//         onClick={() => {
//           props.dispatch(removeExpense({ id: props.expense.id }));
//           props.history.push('/');
//         }}
//       >Remove</button>
//     </div>
//   );
// };

// Remove expense via dispatch and then redirect to dashboard

const mapStateToProps = (state, props) => {
  console.log(props);
  return {
    expense: state.expenses.find((expense) => (
      // props.match.params.id comes from react-dom-router
      expense.id === props.match.params.id
    ))
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (expense, updates) => dispatch(editExpense(expense, updates)),
    removeExpense: (expense) => dispatch(removeExpense(expense))
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);