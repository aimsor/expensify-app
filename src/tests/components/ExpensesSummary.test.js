import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';

test('should render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenses={[expenses[1]]}
      expensesTotal={selectExpensesTotal([expenses[1]])}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary 
      expenses={expenses}
      expensesTotal={selectExpensesTotal(expenses)}
    />
  );
  expect(wrapper).toMatchSnapshot();
});