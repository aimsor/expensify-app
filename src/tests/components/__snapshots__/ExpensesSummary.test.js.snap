// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
     
    expenses
     totalling 
    $1,300.00
  </h3>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
     
    expenses
     totalling 
    $1,346.95
  </h3>
</div>
`;
